src/rust/engine/dep_inference/src/python/mod.rs

```diff
@@ -322,7 +322,9 @@ impl Visitor for ImportCollector<'_> {
     fn visit_string(&mut self, node: tree_sitter::Node) -> ChildBehavior {
         let range = node.range();
         let text: &str = self.string_at(range);
-        if !text.contains(|c: char| c.is_ascii_whitespace() || c == '\\') {
+        if !text.contains(|c: char| c.is_ascii_whitespace() || c == '\\')
+            && !self.is_pragma_ignored(node.parent().unwrap())
```

Comment:
Aargh. the tests in the python code are failing. What node should I be passing to `self.is_pragma_ignored`?

Reply:
In the Tree-sitter playground, for this simple string:
```python
"." # pants: no-infer-dep
```
The comment is a sibling of the strings's parent in this tree:
```
[module](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [1, 0]
  [expression_statement](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 3]
    [string](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 3]
  [comment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 4] - [0, 25]
```

For a simple assignment:
```python
w = "." # pants: no-infer-dep
```
The comment is a sibling of the string's grandparent
```
[module](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [1, 0]
  [expression_statement](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 7]
    [assignment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 7]
      left: [identifier](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 1]
      right: [string](https://tree-sitter.github.io/tree-sitter/playground#) [0, 4] - [0, 7]
  [comment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 8] - [0, 29]
```

For a strings in a list:
```python
a = [ # pants:
  "asdf",
  "..", # pants: no-infer-dep
  ".", # pants: no-infer-dep
  "qwerty",
]
```
The comment is a directly sibling of the string.
```
[module](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [6, 0]
  [expression_statement](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [5, 1]
    [assignment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [5, 1]
      left: [identifier](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 1]
      right: [list](https://tree-sitter.github.io/tree-sitter/playground#) [0, 4] - [5, 1]
        [comment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 6] - [0, 14]
        [string](https://tree-sitter.github.io/tree-sitter/playground#) [1, 2] - [1, 8]
        [string](https://tree-sitter.github.io/tree-sitter/playground#) [2, 2] - [2, 6]
        [comment](https://tree-sitter.github.io/tree-sitter/playground#) [2, 8] - [2, 29]
        [string](https://tree-sitter.github.io/tree-sitter/playground#) [3, 2] - [3, 5]
        [comment](https://tree-sitter.github.io/tree-sitter/playground#) [3, 7] - [3, 28]
        [string](https://tree-sitter.github.io/tree-sitter/playground#) [4, 2] - [4, 10]
```

For a strings in a dict:
```python
b = {
  "asdf": "adsfasdf",
  "a": "..", # pants: no-infer-dep
  "b": ".", # pants: no-infer-dep
  "qwer": "adsfasdf",
}
```
The comment is a sibling of the string's parent:
```
[module](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [6, 0]
  [expression_statement](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [5, 1]
    [assignment](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [5, 1]
      left: [identifier](https://tree-sitter.github.io/tree-sitter/playground#) [0, 0] - [0, 1]
      right: [dictionary](https://tree-sitter.github.io/tree-sitter/playground#) [0, 4] - [5, 1]
        [pair](https://tree-sitter.github.io/tree-sitter/playground#) [1, 2] - [1, 20]
          key: [string](https://tree-sitter.github.io/tree-sitter/playground#) [1, 2] - [1, 8]
          value: [string](https://tree-sitter.github.io/tree-sitter/playground#) [1, 10] - [1, 20]
        [pair](https://tree-sitter.github.io/tree-sitter/playground#) [2, 2] - [2, 11]
          key: [string](https://tree-sitter.github.io/tree-sitter/playground#) [2, 2] - [2, 5]
          value: [string](https://tree-sitter.github.io/tree-sitter/playground#) [2, 7] - [2, 11]
        [comment](https://tree-sitter.github.io/tree-sitter/playground#) [2, 13] - [2, 34]
        [pair](https://tree-sitter.github.io/tree-sitter/playground#) [3, 2] - [3, 10]
          key: [string](https://tree-sitter.github.io/tree-sitter/playground#) [3, 2] - [3, 5]
          value: [string](https://tree-sitter.github.io/tree-sitter/playground#) [3, 7] - [3, 10]
        [comment](https://tree-sitter.github.io/tree-sitter/playground#) [3, 12] - [3, 33]
        [pair](https://tree-sitter.github.io/tree-sitter/playground#) [4, 2] - [4, 20]
          key: [string](https://tree-sitter.github.io/tree-sitter/playground#) [4, 2] - [4, 8]
          value: [string](https://tree-sitter.github.io/tree-sitter/playground#) [4, 10] - [4, 20]
```

This seems very hairy.